zenodo/modules/metrics/api.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2016, 2017 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Zenodo Metrics API."""
+
+from __future__ import absolute_import
+
+from flask import current_app
+from invenio_accounts.models import User
+from invenio_communities.models import Community
+from invenio_files_rest.models import FileInstance
+
+
+class ZenodoMetric(object):
+    """API class for Zenodo Metrics."""
+
+    metrics_start_date = current_app.config['ZENODO_METRICS_START_DATE']
+
+    @staticmethod
+    def get_users():
+        """Get number of unique zenodo users.
+        """
+        return '1'
+
+    @staticmethod
+    def get_visitors():
+        """Get number of unique zenodo users.
+        """
+        return '1'
+
+    @staticmethod
+    def get_researchers():
+        """Get number of unique zenodo users.
+        """
+        return User.query\
+            .filter(User.confirmed_at >= ZenodoMetric.metrics_start_date) \
+            .filter(User.confirmed_at.isnot(None))\
+            .count()
```

Comment:
```suggestion
        return User.query.filter(
            User.confirmed_at >= ZenodoMetric.metrics_start_date,
            User.confirmed_at.isnot(None),
            User.active.is_(True),
        ).count()
```

- In SQLAlchemy for [`User.query.filter(...)`](https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query) you can pass multiple clauses as arguments to the function call and they will all be `AND`ed together
- We also need the `User.active.is_(True)`, to filter out deleted/banned/inactivated users.
- For general formatting we don't have it officially configured, but going with something like `black` (I used [this webapp](https://black.vercel.app/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AGXAOJdAD2IimZxl1N_WlXnON2nzOzKrgfhIz2E5t5_Av2bG9X9w2E4-Df9LyfMmZ5zwjoT6e8L4NHaQdkp2XqjN3dyAI6gTFaMc5gTAGv5ipoTEHyjW_tZVceCTIqW21BJoHCrjzS0ZDKb-spI9j_2kY4lX3Y16CtRdQpuxmcdr8khMYS7MOcxGotZvFM-9Klxpz4BBUKIti1RZbfqbL69UWIkcRcdrdHYo5_zXJzodndo5_kmjLjqK_snXOVMpRLLz6_KNtqNqtH2-S1GXx3L5Z9dWpn-dxPK2rRjH-SAeRF51Z80SwAAAABuH1YZEKSlrwAB_gGYAwAAFvIfq7HEZ_sCAAAAAARZWg==)) gives decent results.